src/Rule/EachHandler.php

```diff
@@ -23,8 +23,9 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
             throw new UnexpectedRuleException(Each::class, $rule);
         }
 
+        $valueAsArray = $context->getParameter(ValidationContext::PARAMETER_VALUE_AS_ARRAY);
         /** @var mixed $value */
-        $value = $context->getParameter(ValidationContext::PARAMETER_VALUE_AS_ARRAY) ?? $value;
+        $value = $valueAsArray !== null ? $valueAsArray : $value;
```

Comment:
Seems, it's same but with more code and more complicated?

Reply:
Reverted the change.